src/Workspaces/Core/Portable/Workspace/Solution/Checksum.cs

```diff
@@ -64,35 +67,32 @@ public string ToBase64String()
             this.WriteTo(bytes);
             return Convert.ToBase64String(bytes);
 #else
-            unsafe
-            {
-                var data = new byte[HashSize];
-                fixed (byte* dataPtr = data)
-                {
-                    *(HashData*)dataPtr = Hash;
-                }
-
-                return Convert.ToBase64String(data, 0, HashSize);
-            }
+            var bytes = new byte[HashSize];
```

Comment:
>var bytes = new byte[HashSize]; [](http://example.com/codeflow?start=12&length=31)

is stackalloc not available on framework?

Reply:
it's not the stackalloc.  It's that Convert.ToBase64String only takes a byte[] on fx.

Reply:
Even dumber question, would it be possible to keep the unsafe code and point it directly to Data1?

Reply:
yes.  a pointer to 'this' would also be a pointer to Data1.

That said, unsafe isn't buying is anything here.  We're still just assigning our bytes into the 16byte array location.  We can do that safely with spans, effectively just as fast.  

The thing we unfortunately can't avoid (without some Convert alternative) is the allocation of the array.

Reply:
Yeah, I was hoping there was a way to get rid of that allocation by somehow using the address of Data1 to be the byte[] passed into ToBase64String.

Reply:
unfortunately no.  arrays are the ones that are on the heap and represent the full fat object.  It's Span (and related types) that allow you to operate over arrays, or contiguous ranges of memory on the stack/heap, with ease.